openssl/src/x509/mod.rs

```diff
@@ -129,32 +129,34 @@ impl X509StoreContext {
 /// let cert_pem = cert.to_pem().unwrap();
 /// let pkey_pem = pkey.private_key_to_pem().unwrap();
 /// ```
-pub struct X509Generator {
+pub struct X509Generator<'ctx> {
```

Comment:
Another possibility could be to add a separate `ca_sign` method that takes in the CA components if transferring ownership of the CA to the generator won't work.